Clients/GUI/Modals/Currency/CurrencyBar.razor

```diff
@@ -0,0 +1,43 @@
+﻿
+    <p class="badge-dark" style="margin-bottom:1px">CURRENCY RATE</p>
+@if (Rates != null)
+{
+    @foreach (var r in Rates)
+    {
+        <div class="text-center container-fluid" style="white-space:nowrap">
+            <span class="item i-body">@r.Code</span>
+            <span class="item i-body">@r.Value</span>
+        </div>
+    }
+}
+@if (Codes == null || Codes.Count == 0)
+{
+    <div class="spinner-border" role="status">
+        <span class="sr-only">Loading...</span>
+    </div>
+}
+
+@code {
+    [Parameter]
+    public List<string> Codes { get; set; }
+
+    List<ExchangeRate> Rates { get; set; }
+
+    protected override async Task OnInitializedAsync()
```

Comment:
I guess we should handle some exceptions here. For example, when there's some error in the process of getting currency rate from NewsService for some our server-side reason, may be we can add some error text instead of rate or at least avoid crashing the whole app with unhandled exception.

Reply:
Fixed